### PR TITLE
feat: [PIE-2618]: Upgrade Formik - NEW PR

### PR DESCRIPTION
**Pull request by ishantmehtaharness**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by ishantmehtaharness** [comment]
retrigger titlecheck

File: src/modules/30-secrets/components/TextReference/TextReference.tsx [hunk 1]
```diff
@@ -44,11 +44,11 @@ interface TextReferenceProps {
 }
 
 interface FormikTextReference extends TextReferenceProps {
-  formik: FormikContext<any>
+  formik: FormikContextType<any>
   tooltipProps?: DataTooltipInterface
 }
 
-const errorCheck = (name: string, formik?: FormikContext<any>) =>
+const errorCheck = (name: string, formik?: FormikContextType<any>) =>
```
Comment (diptiman-adak) [inline src/modules/30-secrets/components/TextReference/TextReference.tsx:51]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
const errorCheck = (name: string, formik?: FormikContextType<unknown>) =>
```

File: src/modules/75-cd/components/PipelineSteps/DeployEnvStep/DeployEnvStep.tsx [hunk 2]
```diff
@@ -423,7 +423,7 @@ const DeployEnvironmentWidget: React.FC<DeployEnvironmentProps> = ({
           environmentRef: getEnvironmentRefSchema(getString)
         })}
       >
-        {formik => {
+        {(formik: any) => {
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/DeployEnvStep/DeployEnvStep.tsx:426]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
        {(formik: unknown) => {
```

File: src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/FailureStrategyPanel/FailureStrategyPanel.stories.tsx [hunk 3]
```diff
@@ -42,7 +42,7 @@ export const Basic: Story<BasicArgs> = args => {
         })}
         onSubmit={() => void 0}
       >
-        {formik => {
+        {(formik: any) => {
```
Comment (diptiman-adak) [inline src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/FailureStrategyPanel/FailureStrategyPanel.stories.tsx:45]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
        {(formik: unknown) => {
```

File: src/modules/10-common/components/MultiTypeText/MultiTypeText.tsx [hunk 4]
```diff
@@ -39,7 +39,7 @@ export interface MultiTypeTextProps {
   className?: string
   name: string
   label: string | React.ReactElement
-  formik?: FormikContext<any>
+  formik?: FormikContextType<any>
```
Comment (diptiman-adak) [inline src/modules/10-common/components/MultiTypeText/MultiTypeText.tsx:42]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
  formik?: FormikContextType<unknown>
```

File: src/modules/75-cd/components/PipelineStudio/DeployStage/EditStageView/EditStageView.tsx [hunk 5]
```diff
@@ -226,7 +226,7 @@ export const EditStageView: React.FC<EditStageViewProps> = ({
             }}
             validationSchema={Yup.object().shape(getNameAndIdentifierSchema(getString, contextType))}
           >
-            {formikProps => {
+            {(formikProps: any) => {
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineStudio/DeployStage/EditStageView/EditStageView.tsx:229]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
            {(formikProps: unknown) => {
```

File: src/modules/75-ce/components/Connectors/AWSCOConnector/steps/OverviewStep.tsx [hunk 6]
```diff
@@ -142,7 +142,7 @@ const OverviewStep: React.FC<StepProps<ConnectorConfigDTO> & OverviewStepProps>
           })
         })}
       >
-        {formikProps => {
+        {(formikProps: any) => {
```
Comment (diptiman-adak) [inline src/modules/75-ce/components/Connectors/AWSCOConnector/steps/OverviewStep.tsx:145]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
        {(formikProps: unknown) => {
```

File: src/modules/10-common/components/ShellScriptMonaco/ShellScriptMonaco.tsx [hunk 7]
```diff
@@ -136,4 +136,4 @@ export function ShellScriptMonaco(props: ConnectedShellScriptMonacoProps): React
   )
 }
 
-export const ShellScriptMonacoField = connect<ShellScriptMonacoProps>(ShellScriptMonaco)
+export const ShellScriptMonacoField = connect<ShellScriptMonacoProps>(ShellScriptMonaco as any)
```
Comment (diptiman-adak) [inline src/modules/10-common/components/ShellScriptMonaco/ShellScriptMonaco.tsx:139]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
export const ShellScriptMonacoField = connect<ShellScriptMonacoProps>(ShellScriptMonaco as unknown)
```

File: src/modules/75-ci/components/PipelineStudio/BuildStageSpecifications/BuildStageSpecifications.tsx [hunk 8]
```diff
@@ -216,7 +216,7 @@ export default function BuildStageSpecifications({ children }: React.PropsWithCh
           formName="ciBuildStage"
           onSubmit={values => logger.info(JSON.stringify(values))}
         >
-          {formik => {
+          {(formik: any) => {
```
Comment (diptiman-adak) [inline src/modules/75-ci/components/PipelineStudio/BuildStageSpecifications/BuildStageSpecifications.tsx:219]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
          {(formik: unknown) => {
```

File: src/modules/10-common/components/MultiTypeList/MultiTypeList.tsx [hunk 9]
```diff
@@ -39,7 +39,7 @@ export interface MultiTypeListProps {
   multiTextInputProps?: Omit<MultiTextInputProps, 'name'>
   enableConfigureOptions?: boolean
   configureOptionsProps?: MultiTypeListConfigureOptionsProps
-  formik?: FormikContext<any>
+  formik?: FormikContextType<any>
```
Comment (diptiman-adak) [inline src/modules/10-common/components/MultiTypeList/MultiTypeList.tsx:42]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
  formik?: FormikContextType<unknown>
```

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/PropagateWidget/PropagateWidget.tsx [hunk 10]
```diff
@@ -69,7 +69,7 @@ export default function PropagateWidget(props: PropagateWidgetProps): JSX.Elemen
       }}
       enableReinitialize
     >
-      {formik => {
+      {(formik: any) => {
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/PropagateWidget/PropagateWidget.tsx:72]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
      {(formik: unknown) => {
```

File: src/modules/10-common/components/UserGroupsInput/UserGroupsInput.tsx [hunk 11]
```diff
@@ -44,7 +44,7 @@ interface MappedUserGroupData {
 }
 
 export interface FormikUserGroupsInput extends UserGroupsInputProps {
-  formik: FormikContext<any>
+  formik: FormikContextType<any>
```
Comment (diptiman-adak) [inline src/modules/10-common/components/UserGroupsInput/UserGroupsInput.tsx:47]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
  formik: FormikContextType<unknown>
```

File: src/modules/10-common/components/UserGroupsInput/FormMultitypeUserGroupInput.tsx [hunk 12]
```diff
@@ -26,7 +26,7 @@ export interface FormMultiTypeUserGroupInputProps
   extends Omit<ExpressionAndRuntimeTypeProps, 'fixedTypeComponent' | 'fixedTypeComponentProps'> {
   label: string
   tooltipProps?: DataTooltipInterface
-  formik?: FormikContext<any>
+  formik?: FormikContextType<any>
```
Comment (diptiman-adak) [inline src/modules/10-common/components/UserGroupsInput/FormMultitypeUserGroupInput.tsx:29]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
  formik?: FormikContextType<unknown>
```

File: src/modules/10-common/components/MultiTypeSelect/MultiTypeSelect.tsx [hunk 13]
```diff
@@ -36,7 +36,7 @@ export interface MultiTypeSelectProps {
   className?: string
   name: string
   label: string | React.ReactElement
-  formik?: FormikContext<any>
+  formik?: FormikContextType<any>
```
Comment (diptiman-adak) [inline src/modules/10-common/components/MultiTypeSelect/MultiTypeSelect.tsx:39]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
  formik?: FormikContextType<unknown>
```

File: src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/Service.tsx [hunk 14]
```diff
@@ -51,7 +51,7 @@ function Service({
   const isEdit = !!identifier
 
   const updateChangeSource = useCallback(
-    (data: any, formik: FormikContext<MonitoredServiceForm>): void => {
+    (data: any, formik: FormikContextType<MonitoredServiceForm>): void => {
```
Comment (diptiman-adak) [inline src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/Service.tsx:54]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
    (data: unknown, formik: FormikContextType<MonitoredServiceForm>): void => {
```

File: src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/Service.utils.ts [hunk 15]
```diff
@@ -58,7 +58,7 @@ export const onSave = async ({
   formik,
   onSuccess
 }: {
-  formik: FormikContext<any>
+  formik: FormikContextType<any>
```
Comment (diptiman-adak) [inline src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/Service.utils.ts:61]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
  formik: FormikContextType<unknown>
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx [hunk 16]
```diff
@@ -306,7 +306,7 @@ export default function TerraformEditView(
                                 return (
                                   <TFMonaco
                                     name="spec.configuration.spec.backendConfig.spec.content"
-                                    formik={formik}
+                                    formik={formik as any}
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx:309]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
                                    formik={formik as unknown}
```

File: src/modules/75-ce/components/COGatewayConfig/steps/ResourceFulfilment.tsx [hunk 17]
```diff
@@ -92,7 +92,7 @@ const ResourceFulfilment: React.FC<ResourceFulfilmentProps> = props => {
     }
   }
 
-  const handleSpotChange = (formik: FormikContext<any>, val: string) => {
+  const handleSpotChange = (formik: FormikContextType<any>, val: string) => {
```
Comment (diptiman-adak) [inline src/modules/75-ce/components/COGatewayConfig/steps/ResourceFulfilment.tsx:95]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
  const handleSpotChange = (formik: FormikContextType<unknown>, val: string) => {
```

File: src/modules/70-pipeline/components/PipelineSteps/Steps/CustomVariables/CustomVariablesEditableStage.tsx [hunk 18]
```diff
@@ -94,7 +94,7 @@ export function CustomVariablesEditableStage(props: CustomVariableEditableProps)
       validate={debouncedUpdate}
       validationSchema={enableValidation ? getValidationSchema(getString) : undefined}
     >
-      {formik => {
+      {(formik: any) => {
```
Comment (diptiman-adak) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/CustomVariables/CustomVariablesEditableStage.tsx:97]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
      {(formik: unknown) => {
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/InlineVarFile.tsx [hunk 19]
```diff
@@ -90,7 +90,7 @@ const InlineVarFile = (props: InlineVarFileProps) => {
             })
           })}
         >
-          {formikProps => {
+          {(formikProps: any) => {
```
Comment (diptiman-adak) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/InlineVarFile.tsx:93]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
          {(formikProps: unknown) => {
```

File: src/modules/70-pipeline/components/CreatePipelineForm/CreatePipelineForm.tsx [hunk 20]
```diff
@@ -60,7 +60,7 @@ export const CreatePipelineForm: React.FC<CreatePipelineFormProps> = props => {
         })
       }}
     >
-      {formikProps => {
+      {(formikProps: any) => {
```
Comment (diptiman-adak) [inline src/modules/70-pipeline/components/CreatePipelineForm/CreatePipelineForm.tsx:63]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
      {(formikProps: unknown) => {
```

File: src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/components/HealthSourceTableContainer/HealthSourceTableContainer.tsx [hunk 21]
```diff
@@ -38,7 +38,7 @@ export default function HealthSourceTableContainer({
   })
 
   const updateHealthSource = useCallback(
-    (data: any, formik: FormikContext<MonitoredServiceForm>): void => {
+    (data: any, formik: FormikContextType<MonitoredServiceForm>): void => {
```
Comment (diptiman-adak) [inline src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/components/HealthSourceTableContainer/HealthSourceTableContainer.tsx:41]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
    (data: unknown, formik: FormikContextType<MonitoredServiceForm>): void => {
```

File: src/modules/35-connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField.tsx [hunk 22]
```diff
@@ -56,7 +56,7 @@ export interface MultiTypeConnectorFieldConfigureOptionsProps
 }
 export interface MultiTypeConnectorFieldProps extends Omit<ConnectorReferenceFieldProps, 'onChange'> {
   onChange?: ExpressionAndRuntimeTypeProps['onChange']
-  formik?: FormikContext<any>
+  formik?: FormikContextType<any>
```
Comment (diptiman-adak) [inline src/modules/35-connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField.tsx:59]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232251)
```suggestion
  formik?: FormikContextType<unknown>
```

**Pull request closed without merging**